lib/from_honeybee/geometry/aperture.rb

```diff
@@ -59,39 +59,110 @@ def to_openstudio(openstudio_model)
         os_vertices << OpenStudio::Point3d.new(vertex[0], vertex[1], vertex[2])
       end
       reordered_vertices = OpenStudio.reorderULC(os_vertices)
-
-      os_subsurface = OpenStudio::Model::SubSurface.new(reordered_vertices, openstudio_model)
-      os_subsurface.setName(@hash[:identifier])
-
-      # assign the construction if it exists
-      if @hash[:properties][:energy][:construction]
-        construction_identifier = @hash[:properties][:energy][:construction]
-        construction = openstudio_model.getConstructionByName(construction_identifier)
-        unless construction.empty?
-          os_construction = construction.get
-          os_subsurface.setConstruction(os_construction)
-        end
-      end
       
-      # assign the bondary condition object if it's a Surface
-      if @hash[:boundary_condition][:type] == 'Surface'
-        # get adjacent sub surface by identifier from openstudio model
-        adj_srf_identifier = @hash[:boundary_condition][:boundary_condition_objects][0]
-        sub_srf_ref = openstudio_model.getSubSurfaceByName(adj_srf_identifier)
-        unless sub_srf_ref.empty?
-          sub_srf = sub_srf_ref.get
-          os_subsurface.setAdjacentSubSurface(sub_srf)
+      # triangulate subsurface if neccesary
+      triangulated = false
+      final_vertices_list = []
+      matching_os_subsurfaces = []
+      matching_os_subsurface_indices = []
+      if reordered_vertices.size > 4
+      
+        # if this apeture has a matched apeture, see if the other one has already been created
+        # the matched apeture should have been converted to multiple subsurfaces
+        if @hash[:boundary_condition][:type] == 'Surface'
```

Comment:
@chriswmackey I think this new code will handle the case of adjacent interior subsurfaces that require triangulation.  That said it is completely untested.  It would be great if you could make an example JSON to unit test this case.

Reply:
This is now tested, thanks for the test model!